kernel/src/process_standard.rs

```diff
@@ -1921,6 +1943,10 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
     /// to be accessible to the process and to not overlap with the grant
     /// region.
     fn in_app_owned_memory(&self, buf_start_addr: *const u8, size: usize) -> bool {
+        // TODO: On CHERI platforms, it impossible to form syscalls with pointers
+        // that are not in app memory. However, buf_start_addr is not the right
+        // type to make this function always return true. If CapabilityPtr makes it
+        // slightly further, we can skip this check.
```

Comment:
```suggestion
```

What is CHERI and why is it relevant here?

Reply:
No idea! Removed all mention of CHERI, and instead wrote a more generic comment about why it might be a good idea to still have `CapaiblityPtr` making it as far as this function at a later point.